cellxgene_schema_cli/cellxgene_schema/validate.py

```diff
@@ -515,9 +521,32 @@ def _validate_column_dependencies(
         all_rules = []
 
         for dependency_def in dependencies:
+            if "complex_rule" in dependency_def:
+                if "match_ancestors" in dependency_def["complex_rule"]:
+                    rule_def = dependency_def["complex_rule"]["match_ancestors"]
+                    validate_curie_ancestors_vectorized = vectorize(self._validate_curie_ancestors)
+                    ancestor_map = rule_def["ancestors"]
+                    inclusive = rule_def["inclusive"]
+
+                    def filter_df_by_ancestor_match(term_id, ontologies, ancestors):
+                        allowed_ancestors = dict(zip(ontologies, ancestors))
+                        return validate_curie_ancestors_vectorized(term_id, "", allowed_ancestors, inclusive, False)
+
+                    # hack: pandas dataframe query doesn't support Dict inputs
+                    ontology_keys = []
+                    ancestor_list = []
+                    for key, val in ancestor_map.items():
+                        ontology_keys.append(key)
+                        ancestor_list.append(val)
+                    query_exp = f"@filter_df_by_ancestor_match({rule_def['column']}, {ontology_keys}, {ancestor_list})"
+            elif "rule" in dependency_def:
+                query_exp = dependency_def["rule"]
+            else:
+                continue
```

Comment:
I added this else: continue with the idea of not breaking if we add another subcase in the future but I can remove if its too confusing